components/ws-manager/pkg/manager/monitor.go

```diff
@@ -492,9 +492,16 @@ func actOnPodEvent(ctx context.Context, m actingManager, manager *Manager, statu
 			}
 		}
 
-		_, startedDisposal := wso.Pod.Annotations[startedDisposalAnnotation]
-		span.LogKV("startedDisposal", startedDisposal)
-		if terminated && !startedDisposal {
+		var ds *workspaceDisposalStatus
+		if rawDisposalStatus, ok := pod.Annotations[disposalStatusAnnotation]; ok {
+			err := json.Unmarshal([]byte(rawDisposalStatus), ds)
+			if err != nil {
+				log.WithError(err).Errorf("cannnot parse disposalStatusAnnotation: %s", rawDisposalStatus)
+			}
+		}
+
+		span.LogKV("disposalStatusAnnotation", ds)
+		if terminated && (ds == nil || !ds.Status.IsWorking()) {
```

Comment:
If the `ds.Status == DisposalFinished`, will we run the following code?

Reply:
fixed